src/main/kotlin/nomic/data/repositories/games/GameRepository.kt

```diff
@@ -31,6 +33,24 @@ class GameRepository(private val db: Database) : IGameRepository {
         }
     }
 
+    override fun listGames(vararg specifications: IGameSpecification): List<GameModel> {
+        var sequence = db.games
+
+        for (specification in specifications) {
+            sequence = specification.apply(sequence)
+        }
```

Comment:
I don't believe this is as readable as you think it is.  There is a lot of addition logic that has to be held in my head to understand what exactly is happening with this piece of code.